ymer/src/main/java/com/avanza/ymer/MongoPartitionFilter.java

```diff
@@ -38,11 +41,17 @@ public static boolean canCreateFrom(SpaceObjectFilter<?> spaceObjectFilter) {
 	}
 
 	private static BasicDBObject buildFilter(PartitionFilter<?> partitionFilter) {
-		return new BasicDBObject("$or", Arrays.asList(
+		if (loadByShard) {
+			int maxShards = 1024;
+			int[] shards = IntStream.range(0, maxShards).filter(i -> i%partitionFilter.getCurrentPartition() == 0).toArray();
```

Comment:
I believe this will make each partition load a different set of objects than what the GS routing expects.

Let's take an example of some random routingkeys: `1623399` through `1623424`. And let's say that we are loading partition 4 of 11. In this range of routingkeys, the GS routing would then select `1623406` and `1623417`, since those satisfy `_routingKey mod 11 == 4`. For these two objects, `_shard` would be `366` and `377` respectively. But `366 mod 4 == 2` and `377 mod 4 == 1`, so those two shards would not be selected into this `shards` variable. Instead, the 1024 sharding would select a different set of objects than those from the GS routing.

However, if the number of GS partitions is evenly divisible of the max shard count (1024 in this case), then I believe the two calculations would imply the same set of objects for each partition.